research/src/pages/radio-button/radio-button.proposal.mdx

```diff
@@ -0,0 +1,185 @@
+| `disabled`     | `bool`   | `false`       | Prevents the user from interacting with any of the choices. Defaults to false.                   |
+| `form`         | `string` | `null`        | Associates the element with a form in the document whose `id` is this value.                     |
+| `readonly`     | `bool`   | `false`       | Makes it unavailable for user interaction, but the selected choice should still be submitted.    |
+| `required`     | `bool`   | `false`       | Makes it required to have a selected choice.                                                     |
+
+#### Events
+
+...
+
+### Anatomy
+
+<RadioButtonGroupAnatomy />
+
+#### DOM Structure (Implicit Grouping)
+
+The following structure represents a group with the `choices` slot filled with radio buttons (their internal structures are omitted for brevity).
+
+```html
+<oui-radio-button-group name="group-name">
+  <slot name="label">
+    <label part="label"></label>
+  </slot>
+  <slot name="required-indicator">
+    <div part="required-indicator"></div>
+  </slot>
+  <slot name="choices">
+    <div part="choices">
+      <oui-radio-button />
+      <oui-radio-button />
+      <oui-radio-button />
+      ...
+    </div>
+  </slot>
+</oui-radio-button-group>
+```
+
+#### DOM Structure (Explicit Grouping)
+
+The following structure, in contrast, represents a group with an empty `choices` slot. The choices are placed outside the group, providing layout flexibility. Each choice has a `group` attribute with value equal to the radio button group's name, making them explicitly bound as choices of that group.
+
+There is also the possibility of having both implicit and explicit choices.
+
+```html
+<oui-radio-button-group name="group-name">
+  <slot name="label">
+    <label part="label"></label>
+  </slot>
+  <slot name="required-indicator">
+    <div part="required-indicator"></div>
+  </slot>
+  <slot name="choices">
+    <div part="choices">
+      ... (it could also be populated)
+    </div>
+  </slot>
+</oui-radio-button-group>
+
+<oui-radio-button group="group-name" />
+<oui-radio-button group="group-name" />
+<oui-radio-button group="group-name" />
+```
+
+#### Slots
+
+| Slot Name            | Description                                                                      | Fallback Content |
+| -------------------- | -------------------------------------------------------------------------------- | ---------------- |
+| `label`              | Add custom markup for the control's label                                        | Empty            |
+| `required-indicator` | Content to indicate the radio button group is required to have a selected option | '\*'             |
+| `choices`            | Slot for placing the radio buttons                                               | Empty            |
+| `error-message`      | Content of any error messages                                                    | Empty            |
```

Comment:
Is the error message UI something that should be handled by default at the browser level or is that something that is better handled via composition?

Reply:
I'll consider it resolved